### PR TITLE
os/arch/arm/src/amebasmart, os/pm: Refine PM DVFS and fix struct naming mismatch for RTL8730E

### PR DESCRIPTION
**Pull request by edwakuwaku**
- Update DVFS wrapper layer
- Fixed PM timer struct naming mismatch (For PR#6015)

### CODE REVIEW

File: os/include/tinyara/pm/pm.h [hunk 1]
```diff
@@ -635,6 +635,23 @@ enum pm_state_e pm_querystate(int domain);
...
+ * Input Parameters:
+ *   div_lvl - voltage frequency scaling level
+ *
+ * Returned Value:
+ *   None.
+ *
+ *
+ ****************************************************************************/
+void up_set_dvfs(int div_lvl);
+
```
Comment (amogh-samsung) [inline os/include/tinyara/pm/pm.h:654]: Please take a look at the function description for pm_dvfs() in pm_dvfs.c and the header file and make any changes required to make the description accurate.

File: os/arch/arm/src/amebasmart/amebasmart_pmc.c [hunk 2]
```diff
@@ -57,11 +57,15 @@
 #include "FreeRTOS.h"
 #include <task.h>
 #include "freertos_pmu.h"
+#else
+#ifdef CONFIG_PM
```
Comment (amogh-samsung) [inline os/arch/arm/src/amebasmart/amebasmart_pmc.c:61]: from Make.defs

```
ifeq ($(CONFIG_PM),y)
CMN_CSRCS += psci.c amebasmart_pmc.c
```

So do we need to check again?

File: os/arch/arm/src/amebasmart/amebasmart_pmc.c [hunk 3]
```diff
@@ -281,7 +285,8 @@ void SOCPS_SleepCG(void)
 	//pmu_acquire_wakelock(PMU_OS);
 }
 
-void up_set_dvfs(u32 val)
+#ifdef CONFIG_PM_DVFS
```
Comment (amogh-samsung) [inline os/arch/arm/src/amebasmart/amebasmart_pmc.c:288]: From Make.defs

```
ifeq ($(CONFIG_PM),y)
CMN_CSRCS += psci.c amebasmart_pmc.c
```

So do we need to check again?

File: os/arch/arm/src/amebasmart/amebasmart_pmhelpers.c [hunk 4]
```diff
@@ -51,7 +51,7 @@
 #include "timer_api.h"
 
 gtimer_t g_timer1;
-extern struct timer_s g_timer_wakeup;
+extern struct pm_wakeup_timer_s g_timer_wakeup;
```
Comment (amogh-samsung) [inline os/arch/arm/src/amebasmart/amebasmart_pmhelpers.c:54]: Please separate this into a different commit as a fix

Comment (edwakuwaku) [reply os/arch/arm/src/amebasmart/amebasmart_pmhelpers.c:54]: This fix is already separated on an independent commit

File: os/include/tinyara/pm/pm.h [hunk 5]
```diff
@@ -635,6 +635,23 @@ enum pm_state_e pm_querystate(int domain);
  ****************************************************************************/
 void pm_dvfs(int div_lvl);
 
+/****************************************************************************
```
Comment (amogh-samsung) [inline os/include/tinyara/pm/pm.h:638]: If CONFIG_PM is checked for compiling up_set_dvfs(), then it should be checked in the header file too. Otherwise up_set_dvfs gets declared in header file but will never be defined

Comment (edwakuwaku) [reply os/include/tinyara/pm/pm.h:640]: Added as suggested~

File: os/arch/arm/src/amebasmart/amebasmart_pmc.c [hunk 6]
```diff
@@ -57,11 +57,13 @@
 #include "FreeRTOS.h"
```
Comment (amogh-samsung) [inline os/arch/arm/src/amebasmart/amebasmart_pmc.c:57]: Please update the description of this commit to briefly explain the changes made in the commit

**Review by amogh-samsung — COMMENTED** [review]
Could you please update the title of the PR according to the latest pushed commit titles and descriptions

**Review by abhishek-samsung — APPROVED** [review]

**Review by thapav — APPROVED** [review]
LGTM

File: os/include/tinyara/pm/pm.h [hunk 7]
```diff
@@ -633,7 +633,28 @@ enum pm_state_e pm_querystate(int domain);
  *   Incorrect frequency scaling level given as input will lead to an assert.
  *
  ****************************************************************************/
+#ifdef CONFIG_PM_DVFS
```
Comment (r-prabu) [inline os/include/tinyara/pm/pm.h:636]: It would be better if the conditional flags also includes the documentation.

**Comment by amogh-samsung** [comment]
This fixes a naming mismatch of PM wakeup timer structure from #6015

File: os/include/tinyara/pm/pm.h (hunk 7 above)
Comment (edwakuwaku) [reply os/include/tinyara/pm/pm.h]: done~

File: os/arch/arm/src/amebasmart/amebasmart_pmc.c (hunk 6 above)
Comment (edwakuwaku) [reply os/arch/arm/src/amebasmart/amebasmart_pmc.c]: done~

**Review by amogh-samsung — APPROVED** [review]

File: os/arch/arm/src/amebasmart/amebasmart_pmc.c [hunk 8]
```diff
@@ -281,7 +283,8 @@ void SOCPS_SleepCG(void)
 	//pmu_acquire_wakelock(PMU_OS);
 }
 
-void up_set_dvfs(u32 val)
+#ifdef CONFIG_PM_DVFS
+void up_set_dvfs(int val)
```
Comment (sunghan-chang) [inline os/arch/arm/src/amebasmart/amebasmart_pmc.c:287]: You add the prototype in pm.h as below
```
void up_set_dvfs(int div_lvl);
```
But here is `(int val)`.

File: os/include/tinyara/pm/pm.h [hunk 9]
```diff
@@ -616,12 +616,13 @@ int pm_changestate(int domain, enum pm_state_e newstate);
...
 
+#ifdef CONFIG_PM_DVFS
 /****************************************************************************
  * Name: pm_dvfs
  *
  * Description:
- *   This internal function is called to reduce the frequency of operation
- *   of the secondary AP core of AI_Dual chipset.
+ *   This internal function is called to reduce the clock frequency
+ *   of the AP core (CA32) of AI_Dual chipset.
```
Comment (sunghan-chang) [inline os/include/tinyara/pm/pm.h:625]: 1. AI Dual is our internal name. It's not proper to use in public github.
2. I left the comment as it's not proper to add chip-specific API because pm is the common module. With the same reason, `AP core (CA32) of AI_Dual chipset` can be changed as general description.

Comment (amogh-samsung) [reply os/include/tinyara/pm/pm.h:625]: I agree, this function is intended to be used in common for all chipsets later.
So lets not have the description specific to one chipset.

Comment (edwakuwaku) [reply os/include/tinyara/pm/pm.h]: Modified to common description, please check~

**Review by sunghan-chang — APPROVED** [review]
